L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -109,19 +197,174 @@ namespace trackFindingTracklet {
+          std::vector<TTBV> Sphis;
+          std::vector<TTBV> Szs;
+          std::vector<TTBV> SdPhis;
+          std::vector<TTBV> SdZs;
+
+          // Create bit vector (gives us all the slicing etc. )
+          TTBV InputLinkTrack(track.second);
+          // Slice Constructor (const TTBV& ttBV, int begin, int end = 0, bool twos =  false)
+          // Unpack KF track 
+          TTBV Tvalid(     InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm + 1, wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm , false );
+          TTBV Tmatch(     InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm    , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp , false );
+          TTBV TsectorPhi( InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp          , wTz + wTcot + wTinvR + wTphi + wTSe , false );
+          TTBV TsectorEta( InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe                 , wTz + wTcot + wTinvR + wTphi , false );
+          TTBV TphiT(      InputLinkTrack , wTz + wTcot + wTinvR + wTphi                        , wTz + wTcot + wTinvR , true );
+          TTBV TinvR(      InputLinkTrack , wTz + wTcot + wTinvR                                , wTz + wTcot , true );
+          TTBV Tcot(       InputLinkTrack , wTz + wTcot                                         , wTz , true );
+          TTBV TzT(        InputLinkTrack , wTz                                                 , 0 , true );
+
+          int temp_z0 = TzT.val() - ((Tcot.val() * modChosenRofZ));
+
+          // Correction to Phi calcuation depending if +ve/-ve phi sector
+          if (TsectorPhi.val() == 0) {
+            BaseSectorCorr = -UnsignedBaseSector;
+          }
+          else{
+            BaseSectorCorr = UnsignedBaseSector;
+          }
+
+          int temp_phi0 = TphiT.val() - ((TinvR.val()) * modChosenRofPhi) + BaseSectorCorr;
+          
+          int temp_tanL = cotBins[TsectorEta.val()] + Tcot.val();
+        
+          TTBV HitPattern(0,setup_->numLayers());
+
+          int tempchi2rphi = 0;
+          int tempchi2rz   = 0;
+
+          for (int iStub = 0; iStub < setup_->numLayers() - 1; iStub++ ){
+            auto stub = streamsStubs[setup_->numLayers()*iLink+iStub].at(iTrack);
+            TTBV InputLinkStub(stub.second);
+
+            // Unpack and store stub bits  
+            TTBV Svalid( InputLinkStub , wSdZ + wSdPhi + wSz + wSphi + wSr + 1 , wSdZ + wSdPhi + wSz + wSphi + wSr , false );
+            TTBV Sr(     InputLinkStub , wSdZ + wSdPhi + wSz + wSphi + wSr     , wSdZ + wSdPhi + wSz+ wSphi , true );
+            TTBV Sphi(   InputLinkStub , wSdZ + wSdPhi + wSz + wSphi           , wSdZ + wSdPhi + wSz , true );
+            TTBV Sz(     InputLinkStub , wSdZ + wSdPhi + wSz                   , wSdZ + wSdPhi , true );
+            TTBV SdPhi(  InputLinkStub , wSdZ + wSdPhi                         , wSdZ , true );
+            TTBV SdZ(    InputLinkStub , wSdZ                                  , 0 , true );
+
```

Comment:
Please use this class to unpack Stubs KF: https://github.com/cms-L1TK/cmssw/blob/539dc1a77b060116c34b65a5e371467d44443b8b/L1Trigger/TrackerTFP/interface/DataFormats.h#L528-L529